test/phoenix_live_view/integrations/components_test.exs

```diff
@@ -167,6 +167,27 @@ defmodule Phoenix.LiveView.ComponentTest do
                assert_receive {:DOWN, ^ref, _, _, _}
              end) =~ "** (ArgumentError) missing required :id in send_update"
     end
+
+    test "warns if component doesn't exist", %{conn: conn} do
+      {:ok, view, _html} = live(conn, "/components")
+
+      assert ExUnit.CaptureLog.capture_log(fn ->
+               send(view.pid, {:send_update, [{StatefulComponent, id: "nemo", name: "NEW-nemo"}]})
+               render(view)
+               refute_receive {:updated, _}
+             end) =~ "send_update failed because component Phoenix.LiveViewTest.StatefulComponent with ID \"nemo\" does not exist or it has been removed"
+    end
+
+    test "raises if component module is not available", %{conn: conn} do
+      Process.flag(:trap_exit, true)
+      {:ok, view, _html} = live(conn, "/components")
+
+      assert ExUnit.CaptureLog.capture_log(fn ->
+               send(view.pid, {:send_update, [{NonexistentComponent, id: "chris", name: "NEW-chris"}]})
+               ref = Process.monitor(view.pid)
+               assert_receive {:DOWN, ^ref, _, _, _}
+             end) =~ "** (ArgumentError) send_update failed (module NonexistentComponent is not available)"
```

Comment:
We could change this one to the "because …" format as well. I used parentheses to match UndefinedFunctionError, though, so I'm happy leaving it like this. I like the idea of it having a similar shape to another commonly seen "No such module" error.

> (UndefinedFunctionError) function NonexistentModule.hello/0 is undefined (module NonexistentModule is not available)